os/migrating-to-clcs.md

```diff
@@ -0,0 +1,377 @@
+
+Going forward, a new method of provisioning via Container Linux Configs is preferred.
+
+This document details how to convert an existing cloud config into a Container Linux Config. Once a Container Linux Config has been written, it is given to the Config Transpiler which will convert it into an Ignition Config. This Ignition Config can then be provided to a booting machine. For more information on this process, take a look at the [Config Transpiler getting started doc][ct-getting-started].
+
+The etcd and flannel examples shown in this document will use dynamic data in the Container Linux Config (anything looking like this: `{PRIVATE_IPV4}`). Not all types of dynamic data are supported on all cloud providers, and if the machine is not on a cloud provider this feature cannot be used. Please see [here][dynamic-data] for more information.
+
+To see all supported options available in a Container Linux Config, please look at the [Container Linux Config schema][ct-config].
+
+## coreos
+
+### etcd2
+
+In a cloud config, etcd version 2 can be enabled and configured by using the `coreos.etcd2.*` section. As an example of this:
+
+```cloud-config
+#cloud-config
+
+coreos:
+  etcd2:
+    discovery:                   "https://discovery.etcd.io/<token>"
+    advertise-client-urls:       "http://$public_ipv4:2379"
+    initial-advertise-peer-urls: "http://$private_ipv4:2380"
+    listen-client-urls:          "http://0.0.0.0:2379,http://0.0.0.0:4001"
+    listen-peer-urls:            "http://$private_ipv4:2380,http://$private_ipv4:7001"
+```
+
+In a Container Linux Config etcd can also be configured, but a Container Linux Config will use the etcd-member service on a Container Linux machine instead of the etcd2 service like coreos-cloudinit would.  The etcd-member service will download a version of etcd of the user's choosing and run it. This means that in a Container Linux Config both etcd v2 and v3 can be configured.
+
+This is done under the etcd section:
+
+```container-linux-config
+etcd:
+    version: 3.1.6
+```
+
+The version can also be omitted to use a sane default version, that will increase with the OS.
+
+Configuration options in this section can be provided the same way as they were in a cloud config, with the exception of dashes (`-`) being replaced with underscores (`_`) in key names.
+
+```container-linux-config:gce
+etcd:
+  name:                        "{HOSTNAME}"
+  advertise_client_urls:       "{PRIVATE_IPV4}:2379"
+  initial_advertise_peer_urls: "{PRIVATE_IPV4}:2380"
+  listen_client_urls:          "http://0.0.0.0:2379"
+  listen_peer_urls:            "http://{PRIVATE_IPV4}:2380"
+  initial_cluster:             "%m=http://{PRIVATE_IPV4}:2380"
+```
+
+### fleet
```

Comment:
I was aiming to cover all possible sections in a cloud config, but I'd be fine with removing it too. Thoughts @crawford ?